cmake/core-files.cmake

```diff
@@ -163,79 +163,87 @@ set(MBGL_CORE_FILES
     src/mbgl/renderer/cross_faded_property_evaluator.hpp
     src/mbgl/renderer/data_driven_property_evaluator.hpp
-    src/mbgl/renderer/debug_bucket.cpp
-    src/mbgl/renderer/debug_bucket.hpp
-    src/mbgl/renderer/fill_bucket.cpp
-    src/mbgl/renderer/fill_bucket.hpp
-    src/mbgl/renderer/fill_extrusion_bucket.cpp
-    src/mbgl/renderer/fill_extrusion_bucket.hpp
     src/mbgl/renderer/frame_history.cpp
     src/mbgl/renderer/frame_history.hpp
     src/mbgl/renderer/group_by_layout.cpp
     src/mbgl/renderer/group_by_layout.hpp
-    src/mbgl/renderer/line_bucket.cpp
-    src/mbgl/renderer/line_bucket.hpp
     src/mbgl/renderer/paint_parameters.hpp
     src/mbgl/renderer/paint_property_binder.hpp
     src/mbgl/renderer/paint_property_statistics.hpp
     src/mbgl/renderer/painter.cpp
     src/mbgl/renderer/painter.hpp
-    src/mbgl/renderer/painter_background.cpp
-    src/mbgl/renderer/painter_circle.cpp
-    src/mbgl/renderer/painter_clipping.cpp
-    src/mbgl/renderer/painter_debug.cpp
-    src/mbgl/renderer/painter_fill.cpp
-    src/mbgl/renderer/painter_fill_extrusion.cpp
-    src/mbgl/renderer/painter_line.cpp
-    src/mbgl/renderer/painter_raster.cpp
-    src/mbgl/renderer/painter_symbol.cpp
     src/mbgl/renderer/possibly_evaluated_property_value.hpp
     src/mbgl/renderer/property_evaluation_parameters.hpp
     src/mbgl/renderer/property_evaluator.hpp
-    src/mbgl/renderer/raster_bucket.cpp
-    src/mbgl/renderer/raster_bucket.hpp
-    src/mbgl/renderer/render_background_layer.cpp
-    src/mbgl/renderer/render_background_layer.hpp
-    src/mbgl/renderer/render_circle_layer.cpp
-    src/mbgl/renderer/render_circle_layer.hpp
-    src/mbgl/renderer/render_custom_layer.cpp
-    src/mbgl/renderer/render_custom_layer.hpp
-    src/mbgl/renderer/render_fill_extrusion_layer.cpp
-    src/mbgl/renderer/render_fill_extrusion_layer.hpp
-    src/mbgl/renderer/render_fill_layer.cpp
-    src/mbgl/renderer/render_fill_layer.hpp
     src/mbgl/renderer/render_item.hpp
     src/mbgl/renderer/render_layer.cpp
     src/mbgl/renderer/render_layer.hpp
     src/mbgl/renderer/render_light.hpp
-    src/mbgl/renderer/render_line_layer.cpp
-    src/mbgl/renderer/render_line_layer.hpp
     src/mbgl/renderer/render_pass.hpp
-    src/mbgl/renderer/render_raster_layer.cpp
-    src/mbgl/renderer/render_raster_layer.hpp
     src/mbgl/renderer/render_source.cpp
     src/mbgl/renderer/render_source.hpp
     src/mbgl/renderer/render_source_observer.hpp
-    src/mbgl/renderer/render_symbol_layer.cpp
-    src/mbgl/renderer/render_symbol_layer.hpp
     src/mbgl/renderer/render_tile.cpp
     src/mbgl/renderer/render_tile.hpp
-    src/mbgl/renderer/symbol_bucket.cpp
-    src/mbgl/renderer/symbol_bucket.hpp
     src/mbgl/renderer/tile_parameters.hpp
     src/mbgl/renderer/tile_pyramid.cpp
     src/mbgl/renderer/tile_pyramid.hpp
     src/mbgl/renderer/transitioning_property.hpp
     src/mbgl/renderer/update_parameters.hpp
 
+    # renderer/buckets
```

Comment:
👍 on subdirectories for buckets and layer. Can you submit that as an independent PR so that it's easily reviewable independently? Will also help minimize potential merge conflicts as these files are getting touched a lot right now.